bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5029,6 +5029,60 @@ BlackboardArtifact newBlackboardArtifact(int artifact_type_id, long obj_id, Stri
 		}
 	}
 
+	/**
+	 * Creates a new data artifact.
+	 * 
+	 * Inserts a row in the blackboard_artifacts.
+	 * And inserts a row in the tsk_data_artifacts table.
+	 *
+	 * @param artifactType       Data artifact type.
+	 * @param obj_id             Object id of parent.
+	 * @param data_source_obj_id Data source object id.
+	 * @param account            Os Account associated with the artifact. May be
+	 *                           null.
+	 * @param connection         Database connection to use.
+	 *
+	 * @return DataArtifact
+	 *
+	 * @throws TskCoreException
+	 */
+	DataArtifact newDataArtifact(BlackboardArtifact.Type artifactType, long obj_id, long data_source_obj_id, OsAccount account, CaseDbConnection connection) throws TskCoreException {
```

Comment:
@bcarrier, @jayaramcs and @raman-bt...

We have a story on the books about moving artifact related methods from SleuthkitCase (deprecating them, actually, and culling the less useful ones from the refactoring) to Blackboard. Do we want to invest a some time right now in not having a SleuthkitCase.newDataArtifact() method to deprecate in the near future?

Reply:
Yes, this should go into Blackboard.java